src/components/graph/search/GraphSearchDialog.tsx

```diff
@@ -139,14 +152,62 @@ const GraphSearchDialog = (props: Props) => {
 				setTimestamp(getTimestampAsNumber(filteredSearchHistory[0].item));
 				closeModal();
 			}
-
 			if (filteredSearchHistory.length !== 1 && !currentSearchResult) {
 				ac.current = new AbortController();
 				onValueChange(id, ac.current);
 			}
 		}
 	}, [submittedId]);
 
+	React.useEffect(() => {
+		const searchChannels = {
+			previous: submittedTimestamp
+				? api.sse.getEventSource({
+						type: 'event',
+						queryParams: {
+							startTimestamp: submittedTimestamp,
+							resultCountLimit: 1,
+							searchDirection: SearchDirection.Previous,
+						},
+				  })
+				: null,
+			next: submittedTimestamp
+				? api.sse.getEventSource({
+						type: 'event',
+						queryParams: {
+							startTimestamp: submittedTimestamp,
+							resultCountLimit: 1,
+							searchDirection: SearchDirection.Next,
+						},
+				  })
+				: null,
+		};
+
+		const stopSearch = () => {
+			searchChannels.next?.close();
+			searchChannels.previous?.close();
+		};
+
+		const onChannelResponse = (ev: any) => {
+			stopSearch();
+			const data = (ev as MessageEvent).data;
+			const parsedEvent = JSON.parse(data);
+			const id = getItemId(parsedEvent);
+			ac.current = new AbortController();
+			onValueChange(id, ac.current);
```

Comment:
i think you need to call the `onSearchSuccess` function, because `onValueChange` calls one more api request to find an event by id, while you already have this event in the `parsedEvent`